src/post-worker.js

```diff
@@ -1,758 +1,638 @@
-                body.push({
-                    key: key,
-                    value: value
-                });
+function parseAss (content) {
+  let m, format, lastPart, parts, key, value, tmp, i, j, body
+  const sections = []
+  const lines = content.split(/[\r\n]+/g)
+  for (i = 0; i < lines.length; i++) {
+    m = lines[i].match(/^\[(.*)\]$/)
+    if (m) {
+      format = null
+      sections.push({
+        name: m[1],
+        body: []
+      })
+    } else {
+      if (/^\s*$/.test(lines[i])) continue
+      if (sections.length === 0) continue
+      body = sections[sections.length - 1].body
+      if (lines[i][0] === ';') {
+        body.push({
+          type: 'comment',
+          value: lines[i].substring(1)
+        })
+      } else {
+        parts = lines[i].split(':')
+        key = parts[0]
+        value = parts.slice(1).join(':').trim()
+        if (format || key === 'Format') {
+          value = value.split(',')
+          if (format && value.length > format.length) {
+            lastPart = value.slice(format.length - 1).join(',')
+            value = value.slice(0, format.length - 1)
+            value.push(lastPart)
+          }
+          value = value.map(function (s) {
+            return s.trim()
+          })
+          if (format) {
+            tmp = {}
+            for (j = 0; j < value.length; j++) {
+              tmp[format[j]] = value[j]
             }
+            value = tmp
+          }
+        }
+        if (key === 'Format') {
+          format = value
         }
+        body.push({
+          key: key,
+          value: value
+        })
+      }
     }
+  }
 
-    return sections;
+  return sections
 };
 
 self.requestAnimationFrame = (function () {
-    // similar to Browser.requestAnimationFrame
-    var nextRAF = 0;
-    return function (func) {
-        // try to keep target fps (30fps) between calls to here
-        var now = Date.now();
-        if (nextRAF === 0) {
-            nextRAF = now + 1000 / self.targetFps;
-        } else {
-            while (now + 2 >= nextRAF) { // fudge a little, to avoid timer jitter causing us to do lots of delay:0
-                nextRAF += 1000 / self.targetFps;
-            }
-        }
-        var delay = Math.max(nextRAF - now, 0);
-        return setTimeout(func, delay);
-        //return setTimeout(func, 1);
-    };
-})();
-
-var screen = {
-    width: 0,
-    height: 0
-};
+  // similar to Browser.requestAnimationFrame
+  let nextRAF = 0
+  return function (func) {
+    // try to keep target fps (30fps) between calls to here
+    const now = Date.now()
+    if (nextRAF === 0) {
+      nextRAF = now + 1000 / self.targetFps
+    } else {
+      while (now + 2 >= nextRAF) { // fudge a little, to avoid timer jitter causing us to do lots of delay:0
+        nextRAF += 1000 / self.targetFps
+      }
+    }
+    const delay = Math.max(nextRAF - now, 0)
+    return setTimeout(func, delay)
+    // return setTimeout(func, 1);
+  }
+})()
+
+const screen = {
```

Comment:
This causes a conflict with screen object, can you change the name to screenCanvas

```
building:ERROR: Closure compiler run failed:

building:ERROR: /tmp/emscripten_temp_a8fms20c/subtitles-octopus-worker-legacy.js.pp.js.mem.js.jso.js.jso.js.jso.js.jso.js:1:4095: ERROR - [JSC_REDECLARED_VARIABLE_ERROR] Illegal redeclared variable: screen
```

Reply:
https://github.com/libass/JavascriptSubtitlesOctopus/blob/master/src/post-worker.js#L431
looks like my linter tripped up and changed this to const, I assume emscripten re-defines this, which is weird cuz I expected this to be treated like a global

Reply:
375a943ae4324bed123d8c6db7fb1ec03f785c92
does this fix it?

Reply:
Yes, but my IDE conflict this variable with the window.screen object

Reply:
> Yes, this fixes the build, but now my IDE conflict this variable with the window.screen object

this was defined previously too, and I can see it being used in emscripten, so I'm not touching it